metagraph/src/annotation/binary_matrix/row_sparse_disk/row_sparse_disk.hpp

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "annotation/binary_matrix/base/binary_matrix.hpp"
+#include "common/vectors/bit_vector_adaptive.hpp"
+
+namespace mtg {
+namespace annot {
+namespace binmat {
+
+
+class RowSparseDisk : public BinaryMatrix {
+  public:
+    RowSparseDisk() {}
+    RowSparseDisk(const std::function<void(const RowCallback &)> &call_rows,
+                  uint64_t num_columns,
+                  uint64_t num_rows,
+                  uint64_t num_relations);
+
+    uint64_t num_columns() const override { return num_columns_; }
+    uint64_t num_rows() const override { return num_rows_; }
+
+    bool get(Row row, Column column) const override {
+        return Impl(boundary_, int_vector_buffer_params.filename,
+                    int_vector_buffer_params.offset, int_vector_buffer_params.buff_size)
+                .get(row, column);
+    }
+    SetBitPositions get_row(Row row) const override {
+        return Impl(boundary_, int_vector_buffer_params.filename,
+                    int_vector_buffer_params.offset, int_vector_buffer_params.buff_size)
+                .get_row(row);
+    }
+    std::vector<Row> get_column(Column column) const override {
+        return Impl(boundary_, int_vector_buffer_params.filename,
+                    int_vector_buffer_params.offset, int_vector_buffer_params.buff_size)
+                .get_column(num_rows(), column);
+    }
+
+    std::vector<SetBitPositions> get_rows(const std::vector<Row> &rows) const override {
+        return Impl(boundary_, int_vector_buffer_params.filename,
+                    int_vector_buffer_params.offset, int_vector_buffer_params.buff_size)
+                .get_rows(rows);
+    }
+
+    bool load(std::istream &in) override;
+    void serialize(std::ostream &out) const override;
+
+    // number of ones in the matrix
+    uint64_t num_relations() const override {
+        return Impl(boundary_, int_vector_buffer_params.filename,
+                    int_vector_buffer_params.offset, int_vector_buffer_params.buff_size)
+                .num_relations_impl();
+    }
+
+    static void serialize(const std::function<void(binmat::BinaryMatrix::RowCallback)> &call_rows,
+                          const std::string &filename,
+                          uint64_t num_cols,
+                          uint64_t num_set_bits,
+                          uint64_t num_rows);
+
+    void set_buff_size(uint64_t _buff_size) {
```

Comment:
for these kinds of methods, we haven't been using preceding `_`s in variable names.

Reply:
Fixed